vowpalwabbit/cost_sensitive.cc

```diff
@@ -206,17 +206,8 @@ void print_update(vw& all, bool is_test, example& ec, multi_ex* ec_seq, bool act
     if (ec_seq != nullptr)
     {
       num_current_features = 0;
-      // If the first example is "shared", don't include its features.
-      // These should be already included in each example (TODO: including quadratic and cubic).
-      // TODO: code duplication csoaa.cc LabelDict::ec_is_example_header
-      example** ecc = &((*ec_seq)[0]);
-      const example& first_ex = *(*ec_seq)[0];
-
-      v_array<COST_SENSITIVE::wclass> costs = first_ex.l.cs.costs;
-      if (costs.size() == 1 && costs[0].class_index == 0 && costs[0].x < 0)
-        ecc++;
-
-      for (; ecc != &(*ec_seq->cend()); ecc++) num_current_features += (*ecc)->num_features;
```

Comment:
This line should crash since you can't dereference an end iterator. Switched it to use a range iterator since we remove the shared example earlier now